packages/@react-types/radio/src/index.d.ts

```diff
@@ -29,7 +29,7 @@ import {
 } from '@react-types/shared';
 import {ReactElement, ReactNode} from 'react';
 
-export interface RadioGroupProps extends ValueBase<string|null, string>, InputBase, InputDOMProps, Validation<string | null>, LabelableProps, HelpTextProps, FocusEvents {
+export interface RadioGroupProps extends ValueBase<string | number | null, string | number>, InputBase, InputDOMProps, Validation<string | number | null>, LabelableProps, HelpTextProps, FocusEvents {
```

Comment:
I don't think we can infer the type based on what is sent into RadioGroup since Radio's can declare their own value and you could have a mix of strings and numbers.

This leads to an interesting set of changes downstream though because people will now need to change their types from just 'string' to 'string | number' in any controlled usage. See stories I had to edit here.

Do we consider this just a bug fix or is it in the gray area of breaking change?